src/GafferSceneUI/SceneView.cpp

```diff
@@ -1335,77 +1335,21 @@ void SceneView::frame( const GafferScene::PathMatcher &filter, const Imath::V3f
 
 void SceneView::expandSelection( size_t depth )
 {
-	Context::Scope scopedContext( getContext() );
-
 	PathMatcher &selection = const_cast<GafferScene::PathMatcherData *>( m_sceneGadget->getSelection() )->writable();
-	PathMatcher &expanded = expandedPaths()->writable();
 
-	std::vector<string> toExpand;
-	selection.paths( toExpand );
+	PathMatcher newlyExpandedLeaves = ContextAlgo::expandDescendants( getContext(), selection, preprocessedInPlug<ScenePlug>(), depth - 1 );
```

Comment:
Still not positive on the -1 +1 depth stuff, but removing them, here is the failure:

```python
======================================================================
FAIL: testExpandedPaths (GafferSceneUITest.ContextAlgoTest.ContextAlgoTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/andrewk/apps/gaffer/0.33.2.0dev/cent7.x86_64/cortex/9/gaffer/python/GafferSceneUITest/ContextAlgoTest.py", line 122, in testExpandedPaths
    self.assertEqual( GafferSceneUI.ContextAlgo.getExpandedPaths( context ), GafferScene.PathMatcher( [ "/", "/A" ] ) )
AssertionError: GafferScene.PathMatcher( ['/'] ) != GafferScene.PathMatcher( ['/', '/A'] )
----------------------------------------------------------------------
```

Recall that the scene setup in ContextAlgoTest is slightly different:

```
# A
# |__B
#    |__D
#    |__E
# |__C
#    |__F
#    |__G
```

```python
GafferSceneUI.ContextAlgo.setExpandedPaths( context, GafferScene.PathMatcher( [ "/" ] ) )
self.assertEqual( GafferSceneUI.ContextAlgo.getExpandedPaths( context ), GafferScene.PathMatcher( [ "/" ] ) )
newLeafs = GafferSceneUI.ContextAlgo.expandDescendants( context, GafferScene.PathMatcher( [ "/" ] ), A["out"], depth = 1 )
self.assertEqual( GafferSceneUI.ContextAlgo.getExpandedPaths( context ), GafferScene.PathMatcher( [ "/", "/A" ] ) ) # this line fails
self.assertEqual( newLeafs, GafferScene.PathMatcher( [ "/A/B", "/A/C" ] ) )
```